paddle/fluid/pybind/auto_parallel_py.cc

```diff
@@ -634,5 +540,131 @@ void BindAutoParallel(py::module *m) {
   });
 }
 
+static void parse_tensors(PyObject *obj,
+                          phi::distributed::InferSpmdContext *ctx,
+                          const size_t arg_pos) {
+  Py_ssize_t len = PyList_Size(obj);
+  VLOG(6) << "args indx: [" << arg_pos << "] input vector of ["
+          << static_cast<size_t>(len) << "] tensors.";
+  paddle::small_vector<phi::distributed::DistMetaTensor,
+                       phi::kInputSmallVectorSize>
+      ins;
+  ins.reserve(static_cast<size_t>(len));
+  for (Py_ssize_t i = 0; i < len; i++) {
+    DistTensorSpec in = py::cast<DistTensorSpec>(PyList_GetItem(obj, i));
+    VLOG(6) << "Vector emplace_back DistTensorSpec: " << in.to_string();
+    ins.emplace_back(phi::distributed::DistMetaTensor(
+        phi::make_ddim(in.shape()), in.dist_attr()));
+  }
+  ctx->EmplaceBackInputs(ins);
+}
+
+static void parse_tensor(PyObject *obj,
+                         phi::distributed::InferSpmdContext *ctx,
+                         const size_t arg_pos) {
+  VLOG(6) << "args indx: [" << arg_pos << "] input one tensor.";
+  DistTensorSpec in = py::cast<DistTensorSpec>(obj);
+  VLOG(6) << "DistTensorSpec: " << in.to_string();
+  ctx->EmplaceBackInput(phi::distributed::DistMetaTensor(
```

Comment:
Maybe we can consider unify DistMetaTensor and DistTensorSpec in the future.

Reply:
yes, that is on the list